### PR TITLE
Add WebSocketServiceHandler

### PR DESCRIPTION
**Pull request by minwoox**
I need to add `WebSocketServiceHandler` and refactor `WebSocketHandler` before releasing a new version.

### CODE REVIEW

**Review by trustin — APPROVED** [review]
:+1:

File: core/src/main/java/com/linecorp/armeria/common/websocket/WebSocketHandler.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+@UnstableApi
+@FunctionalInterface
+public interface WebSocketHandler<T extends RequestContext> {
+
+    /**
+     * Handles the incoming {@link WebSocket} and returns {@link WebSocket} created via
+     * {@link WebSocket#streaming()} to send {@link WebSocketFrame}s.
+     */
+    WebSocket handle(T ctx, WebSocket in);
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/common/websocket/WebSocketHandler.java:36]: Side note: It is okay at the moment. We may want to change the type `WebSocket` into a request-specific version and a response-specific one later to provide a higher abstraction for writing and reading WebSocket frames.
```java
public interface WebSocketHandler<T extends RequestContext, 
                                  I extends WebSocket, 
                                  O extends WebSocket> {
   ...
}
```

WebSocket is in the early stage and it is not yet known how this API will evolve. Some breaking changes should be acceptable.

**Review by ikhoon — APPROVED** [review]
Thanks!

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Pull request merged**